web/app/js/components/TopModule.jsx

```diff
@@ -253,6 +261,14 @@ class TopModule extends React.Component {
     });
   }
 
+  updateNeighborsFromTapData = (source, sourceLabels) => {
+    // store this outside of state, as updating the state upon every websocket event received
+    // is very costly and causes the page to freeze up
+    let resourceType = _isNil(this.props.query.resource) ? "" : this.props.query.resource.split("/")[0];
+    this.unmeshedSources = processNeighborData(source, sourceLabels, this.unmeshedSources, resourceType);
+    if (this.props.updateUnmeshedSources) {this.props.updateUnmeshedSources(this.unmeshedSources);}
```

Comment:
you don't need this check here, since the current default value of this func is `_noop` so it won't do anything. but also, if we make this prop required, we won't need to worry about checking it here!

Reply:
ah that makes sense :+1: